extensions/wikia/AdEngine/js/utils/scriptLoader.js

```diff
@@ -18,16 +18,22 @@ define('ext.wikia.adEngine.utils.scriptLoader', [
 		return script;
 	}
 
-	function loadAsync(src, node, type) {
-		return loadScript(src, type, true, node);
+	function loadAsync(src, options) {
+		options = options || {};
+		options.isAsync = true;
+
+		return loadScript(src, options);
 	}
 
-	function loadScript(src, type, isAsync, node) {
-		return new Promise(function (resolve, reject) {
-			var script = createScript(src, type, isAsync, node);
-			script.onload = resolve;
-			script.onerror = reject;
-		});
+	function loadScript(src, options) {
+		options = options || {};
+
+		var script = createScript(src, options.type, options.isAsync, options.node);
+
+		script.onload = options.onLoad;
+		script.onerror = options.onError;
```

Comment:
What will happen if `options.onLoad` and `options.onError` are `undefined`? It won't break anything on the page for a user but we won't know there is an error, right?

Reply:
It won't break anything, because it's `undefined` by default.